src/lib/menu/menu-trigger.ts

```diff
@@ -341,7 +343,8 @@ export class MatMenuTrigger implements AfterContentInit, OnDestroy {
       const posX: MenuPositionX = change.connectionPair.overlayX === 'start' ? 'after' : 'before';
       const posY: MenuPositionY = change.connectionPair.overlayY === 'top' ? 'below' : 'above';
 
-      this.menu.setPositionClasses(posX, posY);
+      // Note that the position changes may be outside the Angular zone
+      this._ngZone.run(() => this.menu.setPositionClasses(posX, posY));
```

Comment:
Does `markForCheck` in the `ChangeDetectorRef` not work for this situation?

Reply:
I thought `detectChanges` might work, but no.

The problem is that `onPositionChange` sometimes emits within and sometimes outside NgZone due to the scroll dispatcher. I don't have a complete grasp on it, and this might be better if `onPositionChange` always just emitted from within the zone instead of trying to bring it back in at the component level.

@crisbeto can you weigh in?

Reply:
Ideally we should be consistent in terms of where the event is being emitted, however with the way things are set up at the moment (`onPositionChange` isn't really a "change" event but more of an "update" event since it emits even if the position didn't change) we'd end up hitting the zone for all scroll events. We can address it better once the new connected positioning is in.

Reply:
It would be better at least to only enter the zone when `posX` or `posY` change